mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,225 @@
+from enum import Enum
+import json
+from pathlib import Path
+from pydantic import BaseModel, validator, parse_obj_as
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List, Dict, Any
+import yaml
+
+from mlflow.exceptions import MlflowException
+from mlflow.gateway.constants import PROVIDERS
+from mlflow.gateway.utils import is_valid_endpoint_name, check_configuration_route_name_collisions
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class Provider(str, Enum):
+    UNSPECIFIED_PROVIDER = "custom"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    DATABRICKS_SERVING_ENDPOINT = "databricks_serving_endpoint"
+    MLFLOW = "mlflow"
+
+
+class RouteType(str, Enum):
+    CUSTOM = "custom"
+    LLM_V1_COMPLETIONS = "llm/v1/completions"
+    LLM_V1_CHAT = "llm/v1/chat"
+
+
+class OpenAIConfig(BaseModel):
+    openai_api_key: Optional[str] = None
+    openai_api_key_env_var: Optional[str] = None
```

Comment:
This is the materialized config for the route, so I don't think we need a separate `openai_api_key_env_var` entry here. IMO it should just be `openai_api_key`, and its the responsibility of the YAML config parsing logic to resolve `openai_api_key` from an environment variable, file path, etc.

Can we use syntax like `$ENV_VAR_NAME` to denote an environment variable in the YAML Conf and use URIs to indicate that the value lives in a remote file?

e.g.

```
routes:
    - route1:
       ...
       openai_config:
           openai_api_key: $MY_ENV_VAR
```

or 

```
routes:
    - route1:
       ...
       openai_config:
           openai_api_key: <actual_api_key_value>
```

or

```
routes:
    - route1:
       ...
       openai_config:
           openai_api_key: /tmp/path/to/key/file
```

Reply:
fantastic idea. Implemented and will push shortly!

Reply:
added some tests to validate the behavior of all 3 modes